src/Microsoft.Azure.WebJobs.Host/Executors/DefaultStorageAccountProvider.cs

```diff
@@ -15,6 +16,7 @@ internal class DefaultStorageAccountProvider : IStorageAccountProvider
         private readonly IStorageCredentialsValidator _storageCredentialsValidator;
         private readonly IStorageAccountParser _storageAccountParser;
         private readonly IServiceProvider _services;
+        private readonly ConcurrentDictionary<string, Task<IStorageAccount>> _accounts = new ConcurrentDictionary<string, Task<IStorageAccount>>();
```

Comment:
Thanks for the alternative code, very clear as well.

I'd love to have a think over moving the validation out of this code path anyway in a later PR, so we can remove all the async Task.Run in Blob Table & Queue bindings.

I don't know that the validation needs to occur at the first storage call - we could fire it off and correlate back in the event of errors, operating under the default assumption that everything is correct.